src/proposals/proposalTypes/GovernanceProposal.sol

```diff
@@ -0,0 +1,141 @@
+        token.delegate(proposerAddress);
+    }
+
+    /// @notice push a Governance proposal action
+    /// @param target the target contract
+    /// @param value msg.value
+    /// @param description description of the action
+    /// @param data calldata
+    function _pushGovernanceAction(address target, uint256 value, string memory description, bytes memory data)
+        internal
+    {
+        actions.push(GovernanceAction({target: target, value: value, description: description, data: data}));
+    }
+
+    /// @notice push an action to the Governance proposal with a value of 0
+    /// @param target the target contract
+    /// @param description description of the action
+    /// @param data calldata
+    function _pushGovernanceAction(address target, string memory description, bytes memory data) internal {
+        _pushGovernanceAction(target, 0, description, data);
+    }
+
+    /// @notice Simulate governance proposal
+    /// @param timelockAddress address of the timelock
+    /// @param governorAddress address of the artemis governor
+    /// @param proposerAddress address of the proposer
+    function _simulateGovernanceActions(
+        address timelockAddress,
+        address governorAddress,
+        address proposerAddress,
+        string memory description
+    ) internal {
+        require(actions.length > 0, "Empty governance operation");
+
+        /// @dev deal and delegate, so the proposal can be simulated end-to-end
+        Addresses addresses = new Addresses();
+        Well well = Well(payable(addresses.getAddress("WELL")));
+        _deal(address(well), address(this), 100_000_000e18);
+        _delegate(address(this), well);
+
+        /// @dev skip ahead
+        vm.roll(block.number + 1000);
+
+        /// @dev loop through the struct and prep the data for the governance proposal
+        address[] memory targets = new address[](actions.length);
+        uint256[] memory values = new uint256[](actions.length);
+        string[] memory signatures = new string[](actions.length);
+        bytes[] memory calldatas = new bytes[](actions.length);
+        for (uint256 i = 0; i < actions.length; i++) {
+            targets[i] = actions[i].target;
+            values[i] = actions[i].value;
+            signatures[i] = "";
+            calldatas[i] = actions[i].data;
+        }
+
+        MoonwellArtemisGovernor governor = MoonwellArtemisGovernor(governorAddress);
+        bytes memory encoded = abi.encodeWithSignature(
+            "propose(address[],uint256[],string[],bytes[],string)", targets, values, signatures, calldatas, description
+        );
+
+        /// @dev output
+        if (DEBUG) {
+            console.log("Governance proposal with", actions.length, (actions.length > 1 ? "actions." : "action."));
+            emit log_bytes(encoded);
+        }
+
+        (bool success, bytes memory data) = address(governor).call{value: 0}(encoded);
+        require(success);
```

Comment:
could you add a revert reason here? maybe getting the error from `data` would make debugging easier